File: 프로그래머스/lv1/17682. ［1차］ 다트 게임/［1차］ 다트 게임.py
def solution(dartResult):
    answer = 0
    n=[]
    for i,c in enumerate(dartResult):
        if c.isalpha():
            if dartResult[i-2].isdigit(): 
                    n.append(int(dartResult[i-2:i]))
            else : n.append(int(dartResult[i-1]))
        if c=='D':
            n[-1]=n[-1]**2
        if c=='T':
            n[-1]=n[-1]**3
        elif c=='*':
            n[-1]=n[-1]*2
            if len(n)>=2: n[-2]=n[-2]*2
        elif c=='#':
            n[-1]=-n[-1]
            
    return sum(n)

File: 프로그래머스/1/17682. ［1차］ 다트 게임/［1차］ 다트 게임.java
import java.util.*;
class Solution {
    public int solution(String dartResult) {
        int answer = 0;
        int[] cnt=new int[4];
        int index=0;
        for(int i=0;i<dartResult.length();i++){
           
            char c=dartResult.charAt(i);
            if(Character.isDigit(c)){
                index++;
                if(Character.isDigit(dartResult.charAt(i+1))) {
                    cnt[index]=10;
                  i+=1;
                }
                else cnt[index]=c-'0';
            }else if(Character.isLetter(c)){
                
                if(c=='S') cnt[index]=(int)Math.pow(cnt[index],1);
                else if(c=='D') cnt[index]=(int)Math.pow(cnt[index],2);
                else if(c=='T') cnt[index]=(int)Math.pow(cnt[index],3);
            }else{
                if(c=='*'){
                    cnt[index-1]*=2;
                    cnt[index]*=2;
                }else if(c=='#'){
                    cnt[index]*=-1;
                }
            }
        }
        
        for(int i:cnt) answer+=i;
        
        return answer;
    }
}